GoogleForCreators/web-stories-wp
# Editor: Select canvas elements when they are focused

## Context

I think this has been around for a while. Looking at staging it’s there too, so I tracked down the commit. It looks like it started with [this PR](https://github.com/GoogleForCreators/web-stories-wp/pull/10388).

Now when tabbing out of the library, the canvas receives focus. However - there is no event handle on the background frame that sets that element as the 'selected element' in the story.

This is an example of one approach to fix the perceived loss of focus.

## Summary

This 

## Relevant Technical Choices

I think there's two approaches here that would both solve the problem:
1. (as demonstrated in this PR) we can add event handlers to the frame that catch the focus and select the element when it is focused. This has the benefit of being able to change focus from elements using `Tab`. However, this may not be a good thing, so would love your feedback.
2. we can focus the canvas container using `useFocusCanvas`. This has the benefit of working the same way the editor did before.

## To-do

n/a

## User-facing changes

|Before|After|
|--|--|
|![focus-hide](https://user-images.githubusercontent.com/22185279/157112917-176e2ced-a263-46d3-bd39-89ecaaabb852.gif)|![tabbin](https://user-images.githubusercontent.com/22185279/157112635-0908dfb4-7fce-4a3c-ad7a-239a98d0c029.gif)|


## Testing Instructions






This PR can be tested by following these steps:

1. Add an element to the canvas
2. Focus something in the library (any tab)
3. Press `Tab` until you get to the canvas
4. Should see the background focused
5. Every time you press `Tab` from there, if the next focused target is an element on the canvas then that element should be set as 'selected'
6. Once you tab out of the canvas, the previously selected element should remain selected


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10806
